src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -395,6 +407,8 @@ protected final void finalizeGUI(boolean wrapInScrollpane) {
 			openInEditingMode(generatableElement);
 		}
 
+		SwingUtilities.invokeLater(() -> elementUpdateListener.registerUI(centerComponent));
```

Comment:
Why do you use invokelater here?

Reply:
Invoking it directly triggers the listener before the GUI is fully initialized. I even tried to move it at the very end of this method (and, iirc, use ThreadUtil), but it didn't help me.

Reply:
Invoke later will not guarantee this always either, it is asynchronous call

Reply:
You also call registration a few lines above in some cases

Reply:
At least it was always working for me so far :man_shrugging:
>You also call registration a few lines above in some cases

WDYM? Constructor initializer? Tab listener?

Reply:
Sorry, my bad, disregard my last comment.

Still, you can't rely for it to work by making it asynchronous. Figure out what even is being fired and by which component instead, don't just do blind guesses

Reply:
It's fired by `reloadDataLists()`, I'm working on it.

Reply:
:heavy_check_mark: Fixed likely